src/ScriptBuilder/Saga/Writers/OracleSagaScriptWriter.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using NServiceBus.Persistence.Sql.ScriptBuilder;
+
+class OracleSagaScriptWriter : ISagaScriptWriter
+{
+    TextWriter writer;
+    SagaDefinition saga;
+    string tableName;
+
+    public OracleSagaScriptWriter(TextWriter textWriter, SagaDefinition saga)
+    {
+        writer = textWriter;
+        this.saga = saga;
+        if (saga.TableSuffix.Length > 27)
+        {
+            throw new Exception($"Saga '{saga.TableSuffix}' contains more than 27 characters, which is not supported by SQL persistence using Oracle. Either disable Oracle script generation using the SqlPersistenceSettings assembly attribute, shorten the name of the saga, or specify an alternate table name by overriding the SqlSaga's TableSuffix property.");
+        }
```

Comment:
> not supported by SQL persistence using Oracle

Or by Oracle itself? I think it's a bit confusing from the user perspective that they need to be aware that the overall Oracle limit is 30 and in saga we take 3 chars, but in other tables we might take 2. Could we say in all places that the overall limit is 30 and show the full name? People will figure out probably which part is "fixed", but won't have to remember a few different numbers.

> Either disable Oracle script generation using the SqlPersistenceSettings assembly attribute

I'm not clear on how does that solve the problem?